common/mas_knowledge_utils/ontology_query_interface.py

```diff
@@ -245,12 +246,14 @@ def get_all_subjects_and_objects(self, prop):
         else:
             raise ValueError('"{0}" does not exist as a property in the ontology!'.format(prop))
 
-    def get_property_domain_range(self, prop):
+    def get_property_domain_range(self, prop, domain_ns=None, range_ns=None):
```

Comment:
I think passing these as arguments defeats the purpose of the function a bit since these are not known in advance. But a good catch that this case needs to be handled.

I think a better way would be to extend `__extract_class_name` to check the class name and, if this is one of the predefined classes, add the appropriate prefix.

Reply:
True. I fixed it in a slightly different way by modifying the `__extract_class_name`. This function now initially checks if the rdf_class is a URL, if not it assumes it contains a namespace and then extracts the class_name using the delimiter between the `class_prefix` and the `class_name`

Reply:
Yeah, I think that should be fine.